packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -496,4 +652,167 @@ export default class UserStorageController extends BaseController<
       state.isProfileSyncingUpdateLoading = isProfileSyncingUpdateLoading;
     });
   }
+
+  /**
+   * Syncs the internal accounts list with the user storage accounts list.
+   * This method is used to make sure that the internal accounts list is up-to-date with the user storage accounts list and vice-versa.
+   * It will add new accounts to the internal accounts list, update/merge conflicting names and re-upload the results in some cases to the user storage.
+   */
+  async syncInternalAccountsWithUserStorage(): Promise<void> {
```

Comment:
Nice func. Logic is consistent with what was described in PRD and tech doc.

Future - I feel like I want to pull out this logic to keep the controller thin. We can do this later (need to see how bloated this file gets once we add network syncing).

Reply:
Me too, we should definitely extract it like you guys did with set/get storage methods. Will do that in a later PR if that works for you @Prithpal-Sooriya ?

Reply:
Yep later!